frontend/src/metabase/dashboard/components/DashCard/DashCardTabMenu/DashCardTabMenu.unit.spec.tsx

```diff
@@ -0,0 +1,146 @@
+import {
+  createMockDashboard,
+  createMockDashboardOrderedCard,
+} from "metabase-types/api/mocks";
+import {
+  createMockDashboardState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import { getDefaultTab } from "metabase/dashboard/actions";
+import {
+  getDashCardById,
+  getSelectedTabId,
+} from "metabase/dashboard/selectors";
+import { renderWithProviders, screen, waitFor, within } from "__support__/ui";
+import { DashCardTabMenu } from "./DashCardTabMenu";
+
+const DASHCARD = createMockDashboardOrderedCard();
+
+const TEST_DASHBOARD = createMockDashboard({
+  ordered_cards: [DASHCARD],
+});
+
+const TAB_1 = getDefaultTab({ tabId: 1, dashId: 1, name: "Tab 1" });
+const TAB_2 = getDefaultTab({ tabId: 2, dashId: 1, name: "Tab 2" });
+const TAB_3 = getDefaultTab({ tabId: 3, dashId: 1, name: "Tab 3" });
+
+const setup = ({
+  tabs = [TAB_1, TAB_2, TAB_3],
+  selectedTab = TAB_1,
+}: {
+  tabs?: DashboardOrderedTab[];
+  selectedTab?: DashboardOrderedTab;
+}) => {
+  const dashboard = createMockDashboardState({
+    dashboardId: TEST_DASHBOARD.id,
+    selectedTabId: selectedTab.id,
+    dashcards: {
+      [DASHCARD.id]: DASHCARD,
+    },
+    dashboards: {
+      [TEST_DASHBOARD.id]: {
+        ...TEST_DASHBOARD,
+        ordered_cards: TEST_DASHBOARD.ordered_cards.map(c => c.id),
+        ordered_tabs: tabs,
+      },
+    },
+  });
+
+  return renderWithProviders(<DashCardTabMenu dashCardId={DASHCARD.id} />, {
+    storeInitialState: createMockState({
+      dashboard,
+    }),
+  });
+};
+
+describe("DashCardTabMenu", () => {
+  it("when there is only one tab it should not show the menu at all", () => {
+    setup({ tabs: [] });
+
+    expect(screen.queryByText("Move to")).not.toBeInTheDocument();
+  });
+
+  describe("when there are exactly two tabs", () => {
+    it("should show the move card button", () => {
+      setup({ tabs: [TAB_1, TAB_2] });
+
+      expect(getIconButton()).toBeInTheDocument();
+    });
+  });
+
+  describe("when there are more than two tabs", () => {
+    it("should show the move card button", () => {
+      setup({});
+
+      expect(getIconButton()).toBeInTheDocument();
+    });
+
+    it("should show tabs other than the current one as options when hovering the button", async () => {
+      setup({});
+
+      userEvent.hover(getIconButton());
+
+      const menu = await waitFor(() => screen.findByRole("menu"));
+
+      expect(within(menu).queryByText(TAB_1.name)).not.toBeInTheDocument();
+
+      expect(within(menu).getByText(TAB_2.name)).toBeInTheDocument();
+      expect(within(menu).getByText(TAB_3.name)).toBeInTheDocument();
+    });
+  });
+
+  describe("when selecting tab name from the menu", () => {
+    it("should move the card to the selected tab", async () => {
+      const { store } = setup({});
+
+      await clickOnTabMenuItem(TAB_3.name);
+
+      const storeDashcard = getDashCardById(store.getState(), DASHCARD.id);
```

Comment:
All tests below this are checking the redux state, in a future PR I will move them to a upper component to avoid doing it